### PR TITLE
Update developer guide (contacts)

### PR DESCRIPTION
**Pull request by ekweirui**
Updated developer guide for contacts component.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -156,6 +156,60 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+Given below is an example usage scenario and how the delete person mechanism behaves at each step.
+
+Step 1. There currently exists some persons added to a Module.
+
+![DeletePersonStep1ObjectDiagram](images/DeletePersonStep1ObjectDiagram.png)
+
+Step 2. The user executes `delete-person 2` command and this creates a `deletePersonCommand` to delete the 2nd person 
+(that is currently displayed on the GUI) in the address book. Upon execution, `UniqueModuleList#removePersonFromModules
+()` and `UniquePersonList#remove()` are eventually called. `UniqueModuleList#removePersonFromModules()` removes all
```
Comment (Tan-Jia-Rong) [inline docs/DeveloperGuide.md:197]: Notice the use of UniquePersonList#remove() vs the one in @teoyuqi (LogicManager::execute) Perhaps we can discuss on which symbol to adopt in future iteration

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -156,6 +156,60 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+#### Implementation of delete person feature
+The delete person mechanism is facilitated by `DeletePersonCommand` and `DeletePersonCommandParser`.
+`DeletePersonCommandParser` parses the user input in a meaningful way and uses it to create a `DeletePersonCommand`.
+`DeletePersonCommand` then calls the `Model#deletePerson()` operation, which in turn calls
+`AddressBook#removePerson()` to delete a contact in Plannit.
+
+Given below is an example usage scenario and how the delete person mechanism behaves at each step.
+
+Step 1. There currently exists some persons added to a Module.
```
Comment (Tan-Jia-Rong) [inline docs/DeveloperGuide.md:191]:
```suggestion
Step 1. There currently exists some persons added to a Module.
```
Would like to seek some clarification regarding this line. 😅 There seem to be a slight deviation from initial state where contacts can be added even without being associated to a module.

**Review by Tan-Jia-Rong — COMMENTED** [review]
Just some minor details but LGTM in general

File: docs/diagrams/ModelPersonClassDiagram.puml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+
+AddressBook *--> "1" UniquePersonList
+AddressBook *--> "1" UniqueModuleList
+UniqueModuleList --> "~* all" Module
+UniquePersonList --> "~* all" Person
+Person *--> "1" Email
+Person *--> "1" Name
+Person *--> "1" Phone
+Module -right--> "*" Person : contains
```
Comment (cheeheng) [inline docs/diagrams/ModelPersonClassDiagram.puml:38]: Is module containing person a correct representation of the implementation? Here, it looks like the bold triangle is missing. Let's discuss.

**Review by cheeheng — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -156,6 +156,59 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+#### Implementation of delete person feature
+The delete person mechanism is facilitated by `DeletePersonCommand` and `DeletePersonCommandParser`.
+`DeletePersonCommandParser` parses the user input in a meaningful way and uses it to create a `DeletePersonCommand`.
+`DeletePersonCommand` then calls the `Model#deletePerson()` operation, which in turn calls
+`AddressBook#removePerson()` to delete a contact in Plannit.
+
+Given below is an example usage scenario and how the delete person mechanism behaves at each step.
+
+Step 1. There currently exists 3 persons in Plannit, with 2 of them already added to a module.
```
Comment (teoyuqi) [inline docs/DeveloperGuide.md:191]:
```suggestion
Step 1. There currently exists 3 `Person` objects in Plannit, with 2 of them already added to a module.
```

**Review by teoyuqi — APPROVED** [review]
LGTM. Your style of skipping the parsing does look a lot better  :o

**Pull request merged**